src/Microsoft.AspNet.Hosting/Builder/HttpContextFactory.cs

```diff
@@ -9,9 +9,23 @@ namespace Microsoft.AspNet.Hosting.Builder
 {
     public class HttpContextFactory : IHttpContextFactory
     {
+        private IHttpContextAccessor _httpContextAccessor;
+
+        public HttpContextFactory(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
         public HttpContext CreateHttpContext(IFeatureCollection featureCollection)
         {
-            return new DefaultHttpContext(featureCollection);
+            var httpContext = new DefaultHttpContext(featureCollection);
+
+            if (_httpContextAccessor != null)
```

Comment:
Actually we don't support null constructor injection so we may need to adjust that